src/encord_active/cli/metric.py

```diff
@@ -0,0 +1,148 @@
+import importlib.util
+import inspect
+from pathlib import Path
+from typing import Callable, Optional
+
+import rich
+import typer
+from loguru import logger
+
+from encord_active.cli.utils.decorators import ensure_project
+from encord_active.lib.common.utils import fetch_project_meta, update_project_meta
+from encord_active.lib.metrics.metric import Metric, SimpleMetric
+
+metric_cli = typer.Typer(rich_markup_mode="markdown")
+logger = logger.opt(colors=True)
+
+
+@metric_cli.command(name="add", short_help="Add metrics.")
+@ensure_project
+def add_metrics(
+    metric_path: Path = typer.Argument(
+        ..., help="Path to the python module where the metric resides.", exists=True, dir_okay=False
+    ),
+    metric_name: Optional[list[str]] = typer.Argument(None, help="Name of the metric. Can be used multiple times."),
+    target: Path = typer.Option(Path.cwd(), "--target", "-t", help="Path to the target project.", file_okay=False),
+):
+    """
+    Add metrics to the project.
+
+    Add metrics from:
+    - local source python modules
+
+    If no metric name is provided then all metrics found in the python module will be added to the project.
+    """
+    rich.print("Inspecting provided module ...")
+    metric_names = [] if metric_name is None else metric_name
+    metric_names_set = set(metric_names)
+    expanded_metric_path = metric_path.expanduser().resolve()
+
+    project_meta = fetch_project_meta(target)
+    project_metrics = project_meta.setdefault("metrics", dict())
+
+    found_metrics = get_module_metrics(
+        expanded_metric_path, lambda x: len(metric_names) == 0 or x.__name__ in metric_names_set
+    )
+    if found_metrics is None:
+        rich.print("[red]Error: Provided module path doesn't comply with expected format. Check the logs.[/red]")
+        return
+    found_metrics_set = set(found_metrics)
+
+    # add all metrics from the python module if the input doesn't contain any metric name.
+    if len(metric_names) == 0:
+        metric_names = found_metrics.copy()
+        metric_names_set = found_metrics_set.copy()
+
+    has_conflicts = False
+    for m_name in metric_names:
```

Comment:
Why not iterate the `set` instead of the `list`?

Reply:
To preserve ordering

Reply:
so maybe a dictionary? (the keys are always in order)

Reply:
done, a refactoring made the usage of list + set version not needed as no deletion was required so a dict was indeed perfect for this use case.